aws/data_source_aws_ecr_repository_test.go

```diff
@@ -25,6 +25,8 @@ func TestAccAWSEcrDataSource_ecrRepository(t *testing.T) {
 					resource.TestCheckResourceAttrPair(resourceName, "registry_id", dataSourceName, "registry_id"),
 					resource.TestCheckResourceAttrPair(resourceName, "repository_url", dataSourceName, "repository_url"),
 					resource.TestCheckResourceAttrPair(resourceName, "tags", dataSourceName, "tags"),
+					resource.TestCheckResourceAttrPair(resourceName, "image_scanning_configuration", dataSourceName, "image_scanning_configuration"),
```

Comment:
If I recall correctly, checking the attribute name of a list/set doesn't return anything so this passing will always be a false positive. Instead the number of elements needs to be checked:

```suggestion
					resource.TestCheckResourceAttrPair(resourceName, "image_scanning_configuration.#", dataSourceName, "image_scanning_configuration.#"),
```